src/actions/evaluation-queue.js

```diff
@@ -0,0 +1,41 @@
+import { IdFactory } from '../actions/history-id-generator'
+
+export const evalIdGenerator = new IdFactory()
+
+let evaluationQueue = Promise.resolve()
+let evaluationResolvers = {}
+let queueSize = 0
```

Comment:
This could probably live in the state, but not sure if its necessary. I included it mostly so we can keep track via testing.